core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -353,10 +371,26 @@ private[spark] class SparkSubmit extends Logging {
     }
 
     // Resolve glob path for different resources.
-    args.jars = Option(args.jars).map(resolveGlobPaths(_, hadoopConf)).orNull
-    args.files = Option(args.files).map(resolveGlobPaths(_, hadoopConf)).orNull
-    args.pyFiles = Option(args.pyFiles).map(resolveGlobPaths(_, hadoopConf)).orNull
-    args.archives = Option(args.archives).map(resolveGlobPaths(_, hadoopConf)).orNull
+    if (isKubernetesCluster) {
+      // Skip dependencies we will handle at the K8s backend.
```

Comment:
Because in K8s as in Mesos we have uris for files targeting the fs in the container and at the machine where the submission was done in cluster mode. So at submit time I need to know what is what. Right now we have local:// but also file:// that resolves in the container AFAIK. So client makes explicit that that this uri is target the submission machine fs.

Reply:
> we have uris for files targeting the fs in the driver container and at the machine where the submission was done in cluster mode

That has existed in Spark forever.

- file: URIs (or no scheme): refers to launcher file system
- local: URIs: refers to the local file system of containers

You shouldn't add another one.

Reply:
Ok then I can remove the `client ://` thing I didnt want that either.